### PR TITLE
Flink job changes for adding content metrics for sourcing identifiers

### PR DESCRIPTION
**Pull request by ishawakankar**
Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

### Type of change

Please choose appropriate options.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### How Has This Been Tested?

Please describe the tests that you ran to verify your changes in the below checkboxes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Ran Test A
- [ ] Ran Test B

**Test Configuration**:
* Software versions:
* Hardware versions:

### Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: metrics-data-transformer/src/main/resources/metrics-data-transformer.conf [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  consumer.parallelism = 1
+  parallelism = 1
+  window.time = 60
+}
+
+timezone = "IST"
+
+service {
+    content.basePath = "https://localhost:9000/action"
+    learner.basePath = "http://localhost:9000/learning-service"
```
Comment (maheshkumargangula) [inline metrics-data-transformer/src/main/resources/metrics-data-transformer.conf:18]: Please use content-service only for system update API. We don't need learning-service.

File: metrics-data-transformer/src/main/scala/org/sunbird/job/metricstransformer/service/MetricsDataTransformerService.scala [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+case class ContentProps(nv: AnyRef, ov: AnyRef)
+
+trait MetricsDataTransformerService {
+  private[this] lazy val logger = LoggerFactory.getLogger(classOf[MetricsDataTransformerService])
+
+  @throws(classOf[InvalidEventException])
+  def processEvent(event: Event, metrics: Metrics, keys: Array[String])(implicit config: MetricsDataTransformerConfig, httpUtil: HttpUtil): Unit = {
+    val identifier = event.nodeUniqueId
+    val url = config.contentServiceBaseUrl + config.contentReadApi + "/" + identifier
+    val response = getSourcingId(url)(config, httpUtil)
```
Comment (maheshkumargangula) [inline metrics-data-transformer/src/main/scala/org/sunbird/job/metricstransformer/service/MetricsDataTransformerService.scala:21]:
```suggestion
    val response = getContent(url)(config, httpUtil)
```

File: metrics-data-transformer/src/main/scala/org/sunbird/job/metricstransformer/service/MetricsDataTransformerService.scala [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+    }
+  }
+  else {
+      logger.info("Learning event skipped, no sourcing identifier found")
+      metrics.incCounter(config.skippedEventCount)
+    }
+
+  }
+
+  def getSourcingId(url: String)(config: MetricsDataTransformerConfig, httpUtil: HttpUtil) : Map[String,AnyRef] = {
```
Comment (maheshkumargangula) [inline metrics-data-transformer/src/main/scala/org/sunbird/job/metricstransformer/service/MetricsDataTransformerService.scala:54]:
```suggestion
  def getContent(url: String)(config: MetricsDataTransformerConfig, httpUtil: HttpUtil) : Map[String,AnyRef] = {
```

File: metrics-data-transformer/src/main/scala/org/sunbird/job/metricstransformer/service/MetricsDataTransformerService.scala [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+    try {
+      updateContentMetrics(contentMetrics, sourcingId)(config, httpUtil)
+      logger.info("Updated content metrics: " + identifier)
+      metrics.incCounter(config.successEventCount)
+    } catch {
+      case ex: Exception =>
+        logger.error("Error while writing content metrics :: " + event.getJson + " :: ", ex)
+        metrics.incCounter(config.failedEventCount)
+        ex.printStackTrace()
+        throw new Exception(s"Error writing metrics for sourcing id: $sourcingId")
```
Comment (maheshkumargangula) [inline metrics-data-transformer/src/main/scala/org/sunbird/job/metricstransformer/service/MetricsDataTransformerService.scala:44]: Please throw the `ex` which is available already. Don't suppress it and create new.

File: metrics-data-transformer/src/main/scala/org/sunbird/job/metricstransformer/task/MetricsDataTransformerConfig.scala [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+  val lpURL: String = config.getString("service.learner.basePath")
+  val contentV3Update = config.getString("content_update_api")
+
+  val defaultHeaders = Map[String, String] ("Content-Type" -> "application/json")
+
+  // Consumers
+  val eventConsumer = "metrics-data-transformer-consumer"
+  val metricsDataTransformerFunction = "metrics-data-transformer-function"
+
+  val metrics: JList[String] = config.getStringList("data.metrics")
```
Comment (maheshkumargangula) [inline metrics-data-transformer/src/main/scala/org/sunbird/job/metricstransformer/task/MetricsDataTransformerConfig.scala:32]: convert it to Scala List using `.asScala`.

**Review by maheshkumargangula — CHANGES_REQUESTED** [review]

File: metrics-data-transformer/src/main/scala/org/sunbird/job/metricstransformer/functions/MetricsDataTransformer.scala [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+import org.apache.flink.api.common.typeinfo.TypeInformation
+import org.apache.flink.configuration.Configuration
+import org.apache.flink.streaming.api.functions.ProcessFunction
+import org.sunbird.job.{BaseProcessFunction, Metrics}
+import org.sunbird.job.metricstransformer.domain.Event
+import org.sunbird.job.metricstransformer.service.MetricsDataTransformerService
+import org.sunbird.job.metricstransformer.task.MetricsDataTransformerConfig
+import org.sunbird.job.util.{ElasticSearchUtil, HttpUtil}
+
+class MetricsDataTransformer(config: MetricsDataTransformerConfig, httpUtil: HttpUtil)
```
Comment (AmiableAnil) [inline metrics-data-transformer/src/main/scala/org/sunbird/job/metricstransformer/functions/MetricsDataTransformer.scala:14]: rename the MetricsDataTransformer to MetricsDataTransformerFunction, as we are following the same other Function in the same repo.

File: metrics-data-transformer/src/main/scala/org/sunbird/job/metricstransformer/functions/MetricsDataTransformer.scala [hunk 7]
```diff
@@ -0,0 +1,47 @@
+package org.sunbird.job.metricstransformer.functions
```
Comment (AmiableAnil) [inline metrics-data-transformer/src/main/scala/org/sunbird/job/metricstransformer/functions/MetricsDataTransformer.scala:1]: rename the package name from functions to function

**Review by manjudr — CHANGES_REQUESTED** [review]
Please add a grafana dashboard for this job, add more assertions and format the code.

**Comment by AmiableAnil** [comment]
This PR is invalid, confirmed with @ishawakankar . Hence closing this PR.

**Pull request closed without merging**